fluss-common/src/main/java/com/alibaba/fluss/lake/lakestorage/PluginLakeStorageWrapper.java

```diff
@@ -0,0 +1,122 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.lake.lakestorage;
+
+import com.alibaba.fluss.config.Configuration;
+import com.alibaba.fluss.exception.TableAlreadyExistException;
+import com.alibaba.fluss.lake.writer.LakeTieringFactory;
+import com.alibaba.fluss.metadata.TableDescriptor;
+import com.alibaba.fluss.metadata.TablePath;
+import com.alibaba.fluss.utils.TemporaryClassLoaderContext;
+import com.alibaba.fluss.utils.WrappingProxy;
+
+/**
+ * A wrapper around {@link LakeStoragePlugin} that ensures the plugin classloader is used for all
+ * {@link LakeCatalog} and {@link LakeStorage} operations.
+ */
+public class PluginLakeStorageWrapper implements LakeStoragePlugin {
+    private final LakeStoragePlugin inner;
+    private final ClassLoader loader;
+
+    private PluginLakeStorageWrapper(final LakeStoragePlugin inner, final ClassLoader loader) {
+        this.inner = inner;
+        this.loader = loader;
+    }
+
+    public static PluginLakeStorageWrapper of(final LakeStoragePlugin inner) {
+        return new PluginLakeStorageWrapper(inner, inner.getClass().getClassLoader());
+    }
+
+    @Override
+    public ClassLoader getClassLoader() {
+        return inner.getClassLoader();
+    }
+
+    @Override
+    public String identifier() {
+        return inner.identifier();
+    }
+
+    @Override
+    public LakeStorage createLakeStorage(Configuration configuration) {
+        try (TemporaryClassLoaderContext ignored = TemporaryClassLoaderContext.of(loader)) {
+            return new ClassLoaderFixingLakeStorage(inner.createLakeStorage(configuration), loader);
+        }
+    }
+
+    static class ClassLoaderFixingLakeCatalog implements LakeCatalog, WrappingProxy<LakeCatalog> {
+
+        private final LakeCatalog inner;
+        private final ClassLoader loader;
+
+        private ClassLoaderFixingLakeCatalog(final LakeCatalog inner, final ClassLoader loader) {
+            this.inner = inner;
+            this.loader = loader;
+        }
+
+        @Override
+        public void createTable(TablePath tablePath, TableDescriptor tableDescriptor)
+                throws TableAlreadyExistException {
+            try (TemporaryClassLoaderContext ignored = TemporaryClassLoaderContext.of(loader)) {
+                inner.createTable(tablePath, tableDescriptor);
+            }
+        }
+
+        @Override
+        public void close() throws Exception {
+            try (TemporaryClassLoaderContext ignored = TemporaryClassLoaderContext.of(loader)) {
+                inner.close();
+            }
+        }
+
+        @Override
+        public LakeCatalog getWrappedDelegate() {
+            return inner;
+        }
+    }
+
+    static class ClassLoaderFixingLakeStorage implements LakeStorage, WrappingProxy<LakeStorage> {
```

Comment:
We don't need to wrap `LakeStorage` for simplicity since the server side will never used it currently.

Reply:
Sorry for misleading. I'm intending to means:
Change 
```
@Override
        public LakeTieringFactory<?, ?> createLakeTieringFactory() {
            try (TemporaryClassLoaderContext ignored = TemporaryClassLoaderContext.of(loader)) {
                return inner.createLakeTieringFactory();
            }
        }
```
to 
```
public LakeTieringFactory<?, ?> createLakeTieringFactory() {
            return inner.createLakeTieringFactory();
        }
```
Not to wrap LakeTieringFactory.